src/octoprint/plugins/gcodeviewer/static/js/viewer/worker.js

```diff
@@ -376,10 +447,23 @@ var doParse = function () {
     // visualizer doesn't actually have a physical offset ;)
     var activeToolOffset = toolOffsets[0];
 
+    // skipUntil preparations
+    var skipUntilFound = false;
+
     var i, j, args;
 
+    // if skipUntil is set, get skipUntilPresent
+    skipUntilPresent = false;
+    if (skipUntil !== undefined && skipUntil !== "") {
+        result = await fetch("/api/plugin/gcodeviewer?path=" + localPath);
```

Comment:
With the above change, this would then be

```suggestion
        result = await fetch("/plugin/gcodeviewer/skipuntilcheck/local/" + localPath);
```